src/openrct2/entity/Balloon.cpp

```diff
@@ -47,6 +47,13 @@ void Balloon::Update()
             {
                 frame = 0;
             }
+
+            if (Collides())
+            {
+                Pop(true);
+                return;
+            }
+
             MoveTo({ x, y, z + 1 });
 
             int32_t maxZ = 1967 - ((x ^ y) & 31);
```

Comment:
Should we make this an `else` part, so that the compiler knows it's only executed if not `Collides`? Also would remove the need for the `return`

Reply:
As far as the compiler is concerned both options produce the same control flow. It's just a matter of style. Some people don't like returning from the middle of a function and other people don't like having too many indents. I find the current code slightly more readable but it's subjective and there's not much in it